### PR TITLE
feat: deposit ist from provision dialog

### PR DESCRIPTION
**Pull request by samsiegart**
fixes https://github.com/Agoric/dapp-inter/issues/244

Screenshot 2024-03-14 at 10 27 40 AM

### CODE REVIEW

**Comment by otoole-brendan** [comment]
@samsiegart the one change I would make (and you might already be doing it but not visible here) is some logic around when a user can click 'Proceed'. I think it's better UX to not let them attempt the transaction by hitting 'Proceed' until they have at least 1IST balance to cover the smart wallet provisioning. So we should only illuminate the button when they have that required balance.

File: src/components/ProvisionSmartWalletNoticeDialog.tsx [hunk 1]
```diff
@@ -55,18 +60,57 @@ const ProvisionSmartWalletNoticeDialog = ({
...
-      its provision which will be deposited into the reserve pool. Click
-      &quot;Proceed&quot; to provision wallet and submit transaction.
-    </span>
+    <>
+      <div>
+        To interact with contracts on the Agoric chain, a smart wallet must be
+        created for your account. As an anti-spam measure, you will need{' '}
+        {smartWalletFeeForDisplay && <b>{smartWalletFeeForDisplay}</b>} to fund
+        its provision which will be deposited into the reserve pool. Click
+        &quot;Proceed&quot; to provision wallet and submit transaction.
```
Comment (raphdev) [inline src/components/ProvisionSmartWalletNoticeDialog.tsx:80]:
```suggestion
        To interact with contracts on the Agoric chain, a smart wallet must be
        created for your account. You will need{' '}
        {smartWalletFeeForDisplay && <b>{smartWalletFeeForDisplay}</b>} to fund
        its provision which will be deposited into the reserve pool. Click
        &quot;Proceed&quot; to provision wallet and submit transaction.
```

**Review by 0xpatrickdev — APPROVED** [review]
Ran into #252 while testing, but appears to be unrelated. LGTM after the copy edit.

**Comment by samsiegart** [comment]
@ivanlei I updated the text in the dialog per @raphdev 's suggestion, lmk if this looks good to you.

**Comment by samsiegart** [comment]
> Ran into #252 while testing, but appears to be unrelated. LGTM after the copy edit.

The breaking change has now been [reverted](https://github.com/Agoric/dapp-inter/pull/253) and this PR has been rebased

**Pull request merged**